server/src/controllers/note.controllers.ts

```diff
@@ -16,17 +29,45 @@ const buildJsonResponse = (data: unknown, key: string) => {
 
 export const deleteNote = async (req: Request, res: Response) => {
   const { noteId } = req.params;
-  const isDeleted = await Note.deleteNoteFromId(noteId);
-  if (isDeleted) {
-    res.status(200).send(`note ${noteId} was succesfully deleted`);
-  } else {
-    res.status(400).send(`note ${noteId} does not exist.`);
+  try {
+    const isDeleted = await Note.deleteNoteFromId(noteId);
+    if (isDeleted) {
+      res.status(200).send(`note ${noteId} was succesfully deleted`);
+    } else {
+      res.status(400).send(`note ${noteId} does not exist.`);
+    }
+  } catch (e) {
+    res.status(500).send('Internal server error.' + (e as Error).message);
   }
 };
 
 export const updateNote = async (req: Request, res: Response) => {
   const { noteId } = req.params;
-  res.send(`update this note  ${noteId}`);
+  const { content, title, lastUpdated } = req.body;
+  if (lastUpdated == null) {
+    res.status(404).send('you must specify the update date');
+    return;
+  }
+
+  //remove nullable field
+
+  let fieldToEdit = Object.fromEntries(
+    Object.entries({ content, title }).filter(([_, v]) => v != null),
+  );
+
+  try {
+    const isUpdated = await Note.updateNote(noteId, {
+      ...fieldToEdit,
+      lastUpdated,
+    });
+    if (isUpdated) {
+      res.status(200).send('Note updated successfully');
```

Comment:
The best practice is to return the note with the updated fields to the client. So you have to change this.